lib/worker.js

```diff
@@ -99,15 +110,46 @@ async function getLinter(filePath) {
     return tslintCache.get(basedir);
   }
 
-  // Initialize the default instance if it hasn't already been initialized
-  loadDefaultTSLint();
-
   if (config.useLocalTslint) {
-    return getLocalLinter(basedir);
+    const localLint = await resolveAndCacheLinter(basedir);
+    if (localLint) {
+      return localLint;
+    }
+  }
+
+  if (fallbackLinter) {
+    tslintCache.set(basedir, fallbackLinter);
+    return fallbackLinter;
+  }
+
+  if (config.useGlobalTslint) {
+    if (config.globalNodePath) {
+      const globalLint = await resolveAndCacheLinter(basedir, config.globalNodePath);
+      if (globalLint) {
+        fallbackLinter = globalLint;
+        return globalLint;
+      }
+    }
+
+    const prefix = await getNodePrefixPath().catch((err) => {
```

Comment:
This should be in a `try`/`catch` block:

```js
let prefix
try {
  prefix = await getNodePrefixPath();
} catch (err) {
  // eslint-disable-next-line no-console
  console.warn(`Attempted to load global tslint, but \`npm get prefix\`
failed. Falling back to the packaged version of tslint. You can specify
your prefix manually in the settings or linter-tslint config file.\n
The error message encountered was:\n\n${err.message}`);
}
```

That's what I meant as part of the "`Promise` syntax", sorry for not making that clear 😉.

Reply:
Ah! The `.catch()`. 😛 Now I understand. Should be easy since that the two promises are separated already.